code/drasil-theory/lib/Theory/Drasil/ConstraintSet.hs

```diff
@@ -36,4 +37,4 @@ instance Express e => Express (ConstraintSet e) where
 
 -- | Smart constructor for building ConstraintSets
 mkConstraintSet :: ConceptChunk -> NE.NonEmpty e -> ConstraintSet e
-mkConstraintSet = CL
+mkConstraintSet c = CL (c +++ "CS") c
```

Comment:
This kind of hack strongly tells me that we need to revamp our `UID` implementation to something more structured. I know MMT uses URIs for doing something similar. I don't like that either, but it is (slightly) better than name concatenation.

I'm okay with keeping the hack for now, as we can easily find it by looking for `+++`.

Reply:
I agree. We need to form some sort of an algebra for our `UID`s. It would be particularly nice if we could use them to track down how chunks were formed, e.g., `data UID = Join (NonEmptyList UID) | Phenomena String | Pushout UID String` (except that this would need a lot more thinking done).

Sorry, what is "MMT"?

> I'm okay with keeping the hack for now, as we can easily find it by looking for +++.
Certainly, I think a better UID system would be a nice goal after the ChunkDBs are updated. This is also related to #2788

Reply:
MMT ([system description paper](https://kwarc.info/people/frabe/Research/rabe_mmtsys_20.pdf) and [site](https://uniformal.github.io/doc/) ) is a very general system for describing theories that might sit in different logics.  It's massively over-engineered (for our purposes), and yet also doesn't contain some crucial knowledge capture mechanisms that we need.

We need to figure out what information should be embedded in our UIDs. Some kind of 'name spaces' comes to mind.

Reply:
Ah, that's an interesting project!

> also doesn't contain some crucial knowledge capture mechanisms that we need.

That's very interesting. I'll need to look into it more.

>  We need to figure out what information should be embedded in our UIDs. Some kind of 'name spaces' comes to mind.

I'm a bit confused by our existing ConceptDomains; I'm having a hard time understanding which "level" we exactly want to call a conceptual domain for something. Do you think a new UID scheme could/should address this?

Perhaps we should move this discussion to a new ticket as well? I'll create a ticket.